salt/modules/cmdmod.py

```diff
@@ -245,6 +245,10 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    # Remove whitespace from strings
+    if isinstance(cmd, six.string_types):
+        cmd = cmd.strip()
+
```

Comment:
But if this is limited to Windows with runas, why put this code at the top of the function where it'll be run on all minions? Why not just put it in [this block](https://github.com/saltstack/salt/pull/36977/files#diff-879e9c967b796ba8a6922815406a96ffR328) so it's confined to the cases where it'll actually be needed?

For that matter, I don't understand why you are splitting and joining again [here](https://github.com/saltstack/salt/pull/36977/files#diff-879e9c967b796ba8a6922815406a96ffR337). This seems like needless extra work.